probe-rs/targets/MIMXRT500.yaml

```diff
@@ -0,0 +1,367 @@
+name: NXP MIMXRT500 Series
+# NOTE: This was originally generated from CMSIS-Pack but subsequently modified
```

Comment:
Do you think it's feasible to implement this functionality into target-gen (not in this PR) in the future, to allow for easy pack refreshes? :)

Reply:
The main thing I had tweaked here was the naming of the memory regions, which in the original pack are pretty opaque and confusing -- it seems like the pack itself was generated from some other source inside NXP and uses auto-generated names.

The only way I can think of to put that in the generator code would be for the generator to have a chip-specific lookup table mapping the pack names to the datasheet names. But I'm skeptical that a future pack release that changes the memory ranges would use the same names anyway, since they seem to be auto-generated as mentioned above.

The other thing I had fixed here is that the core names were inconsistent, making the initial generated YAML not valid at all. It seemed to me like that was a bug in the original pack itself, although it was a while ago I did that part so I might be misremembering.

Reply:
Ok, sounds fair :) Let's leave it as is then :)